tests/ga/test_monitor.py

```diff
@@ -663,36 +681,128 @@ def test_send_extension_metrics_telemetry_for_empty_cgroup(self, patch_report_al
         self.assertEqual(1, patch_poll_all_tracked.call_count)
         self.assertEqual(1, patch_report_all_tracked.call_count)
         self.assertEqual(0, patch_add_event.call_count)
+        self.assertEqual(0, patch_add_metric.call_count)
+        monitor_handler.stop()
+
+    @patch("azurelinuxagent.common.cgroup.MemoryCgroup.get_memory_usage")
+    @patch('azurelinuxagent.common.logger.Logger.periodic_warn')
+    def test_send_extension_metrics_telemetry_handling_memory_cgroup_exceptions(self, patch_periodic_warn,
+                                                                                patch_get_memory_usage, *args):
+        ioerror = IOError()
+        ioerror.errno = 2
+        patch_get_memory_usage.side_effect = ioerror
+
+        CGroupsTelemetry._tracked.append(CpuCgroup("cgroup_name", "/test/path"))
+
+        monitor_handler = get_monitor_handler()
+        monitor_handler.init_protocols()
+        monitor_handler.last_cgroup_polling_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
+        monitor_handler.last_cgroup_report_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
+        monitor_handler.poll_telemetry_metrics()
```

Comment:
Wouldn't we want to assert that the returned metrics from here are empty too?

Reply:
The metrics are returned by poll_all_tracked (From CGroupsTelemetry), and not poll_telemetry_metrics. And thus it won't be returning anything here. 

Had an offline discussion and added newer asserts (to verify sending events is not happening when there's an exception)